File: homework1/src/main.cpp
#include <iostream>
#include <string>
#include <tuple>


struct User
{
    std::string name;
    std::tuple<float,std::string> balance;
    
    void print_user_details()
    {
        std::cout << "Name: " << name << std::endl;
        std::cout << "Balance: " << std::get<0>(balance) << " " << std::get<1>(balance) << std::endl;
    }

    static void accrue_interest(User& user, float rate)
    {
        std::get<0>(user.balance) += (rate / 100.f) * std::get<0>(user.balance);
    }
};

int main()
{
    User user{"John Doe", {1000.f, "SGD"}};
    user.print_user_details();
    User::accrue_interest(user, 10.f);
    User::accrue_interest(user, 10.f);
    User::accrue_interest(user, 10.f);
    User::accrue_interest(user, 10.f);
    User::accrue_interest(user, 10.f);

    user.print_user_details();
    return 0;
}

File: homework1/src/main.rs

struct User
{
    name: String,
    balance: (f32, String),
}

impl User
{
    fn new(name: String, balance: (f32, String)) -> User
    {
        User
        {
            name: name,
            balance: balance,
        }
    }

    fn print_user_details(&self)
    {
        println!("Name: {}", self.name);
        println!("Balance: {} {}", self.balance.0, self.balance.1);
    }

    fn accrue_interest(user: &mut User, rate: f32)
    {
        user.balance.0 += user.balance.0 * (rate / 100.0);
    }
    
}

fn main() {
    println!("Hello, world!");

    let mut user = User::new("John Doe".to_string()
                            , (100.0, "SGD".to_owned()));
    /*let mut user : User = User{
        name: String::from("John Doe"),
        balance: (100.0, String::from("SGD")),
    };*/
    user.print_user_details();
    User::accrue_interest(&mut user, 10.0);
    User::accrue_interest(&mut user, 10.0);
    User::accrue_interest(&mut user, 10.0);
    User::accrue_interest(&mut user, 10.0);
    User::accrue_interest(&mut user, 10.0);
    user.print_user_details();
}
